raysect/core/math/function/function2d/base.pyx

```diff
@@ -162,6 +163,39 @@ cdef class Function2D:
                 return PowScalarFunction2D(<double> a, b)
         return NotImplemented
 
+    def __richcmp__(self, object other, int op):
+        if is_callable(other):
+            if op == Py_EQ:
+                return EqualsFunction2D(self, <Function2D> other)
```

Comment:
Remove the Function2D cast here, otherwise pure python functions will fail. The autowrap in the init will handle the conversion.

Reply:
Thanks, I missed these. I'll remove the casts for all `op`s.